src/serialize/test.rs

```diff
@@ -0,0 +1,48 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde_test::{assert_ser_tokens, Token};
+use serialize;
+
+#[derive(Serialize)]
+struct TestStruct {
+    #[serde(serialize_with = "serialize::hidden_or_not_set")]
+    a: Option<String>,
+}
+
+#[test]
+fn option_not_set() {
+    let s = TestStruct { a: None };
+    assert_ser_tokens(
+        &s,
+        &[
+            Token::Struct {
+                name: "TestStruct",
+                len: 1,
+            },
+            Token::Str("a"),
+            Token::Str("[not set]"),
+            Token::StructEnd,
+        ],
+    );
+}
+
+#[test]
+fn option_set_but_hidden() {
+    let s = TestStruct {
+        a: Some(String::from("sensitive data!")),
+    };
+    assert_ser_tokens(
+        &s,
+        &[
+            Token::Struct {
+                name: "TestStruct",
+                len: 1,
+            },
+            Token::Str("a"),
+            Token::Str("[hidden]"),
+            Token::StructEnd,
+        ],
+    );
+}
```

Comment:
These tests are fine as far as they go but they don't fail if somebody accidentally deletes the `hidden_or_not_set` from one of the sensitive fields, which is the thing I really want to protect against (as it seems far more likely to happen than someone accidentally breaking `hidden_or_not_set` itself.

So the kind of tests I'd really like to see would go in `src/settings/test.rs` and would assert that the serialized string for `Settings` does not contain whatever preset values you assign for the sensitive fields. I've not tried actually writing that out, but it seems like it should be straightforward enough in practice?

Reply:
Makes a lot of sense! I'll check that out.